tests/C-tests/add_matrix_typed.c

```diff
@@ -67,29 +75,43 @@ int TYPED_FUNC(
         bml_print_dense_matrix(N, matrix_precision, dense_row_major, C_dense,
                                0, max_row, 0, max_col);
         LOG_INFO("B = alpha A + beta C \n");
-        bml_print_dense_matrix(N, matrix_precision, dense_row_major, B_dense,
+        bml_print_dense_matrix(N, matrix_precision, dense_row_major, B_1_dense,
                                0, max_row, 0, max_col);
 
+        double expected_trace_norm = 0.0;
         for (int i = 0; i < N * N; i++)
         {
             double expected = alpha * A_dense[i] + beta * C_dense[i];
-            double rel_diff_val = (expected - B_dense[i]) / expected;
+            double rel_diff_val = (expected - B_1_dense[i]) / expected;
             double rel_diff = fabs(rel_diff_val);
             if (rel_diff > REL_TOL)
             {
                 LOG_ERROR
                     ("matrices are not identical; expected[%d] = %e, B[%d] = %e\n",
-                     i, expected, i, B_dense[i]);
+                     i, expected, i, B_1_dense[i]);
                 return -1;
             }
+            expected_trace_norm += expected * expected;
         }
+        expected_trace_norm = sqrt(expected_trace_norm);
```

Comment:
@cnegre The `ellsort` implementation returns the square of the norm. What is the expected output; the norm or the square of the norm?